spec/integration/post_index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require_relative '../rails_helper'
+
+RSpec.describe 'Post index page', type: :system do
+  context 'index page' do
+    it 'User should see the other user\'s profile picture' do
+      visit user_posts_path('105')
+      expect(page).to have_selector('img')
+    end
+
+    it 'User should see the other user\'s username' do
+      visit user_posts_path('105')
+      expect(page).to have_content('Peter')
+    end
+
+    it 'User should see the number of posts of all other users' do
+      visit user_posts_path('105')
+      expect(page).to have_content('Number of posts:')
+    end
+
+    it 'User should see some of a post`body' do
+      visit user_posts_path('105')
+      expect(page).to have_content('An application programming interface')
+    end
+
+    it 'User should see the first comments on a post' do
+      visit user_posts_path('105')
+      expect(page).to have_content('Mark: nice post')
+    end
+
+    it 'User should see how many comments a post has' do
+      visit user_posts_path('105')
+      expect(page).to have_content('Comments:')
+    end
+
+    it 'User should see how many likes a post has' do
+      visit user_posts_path('105')
+      expect(page).to have_content('Likes:')
+    end
+  end
+
+  context 'index page navigation links' do
+    it 'User should be able to check a post\'s show page' do
+      visit user_posts_path('105')
+      click_link 'What does API mean?'
+      expect(page).to have_current_path('/users/105/posts/124')
+    end
+
+    it 'User should see a section for pagination' do
+      visit user_posts_path('105')
+      expect(page).to have_content('pagination')
+    end
+
+    it 'User should see a post title' do
+      visit user_posts_path('105')
+      expect(page).to have_content('What does API mean?')
+    end
+  end
+end
```

Comment:
- [ ] Please fix some of the tests that are failing
<img width="1070" alt="Screen Shot 2023-03-06 at 20 10 40" src="https://user-images.githubusercontent.com/44745855/223196041-e95e9f75-23a9-477d-aa2e-830b21a02587.png">